File: algorithms/longest-common-prefix/LongestCommonPrefix.java
public class LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) return "";
        String result = "";
        for (int i = 0; i < strs[0].length(); i++) {
            char currentChar = strs[0].charAt(i);
            for (int j = 1; j < strs.length; j++) {
                if (i >= strs[j].length() || strs[j].charAt(i) != currentChar) {
                    return result;
                }
            }
            result += currentChar;
        }
        return result;
    }

    public static void main(String[] args) {
        String[] strs = {"flower","flow","flight"};
        String result = new LongestCommonPrefix().longestCommonPrefix(strs);
        System.out.println(result);
    }
}

File: algorithms/longest-common-prefix/longestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        if strs == None or len(strs) == 0:
            return ""
        result  = ""
        for i in range(len(strs[0])):
            currentChar = strs[0][i]
            for j in range(1, len(strs)):
                if i >= len(strs[j]) or strs[j][i] != currentChar:
                    return result
            result += currentChar
        return result
    
str = ["flower","flow","flight"]
result = Solution().longestCommonPrefix(str)
print(str, " -> ", result)